src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchConnectorIT.java

```diff
@@ -0,0 +1,128 @@
+import java.util.Collections;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.storage.StringConverter;
+import org.apache.kafka.test.IntegrationTest;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Category(IntegrationTest.class)
+public class ElasticsearchConnectorIT extends ElasticsearchConnectorBaseIT {
+
+  private static Logger log = LoggerFactory.getLogger(ElasticsearchConnectorIT.class);
+
+  @BeforeClass
+  public static void setupBeforeAll() {
+    container = ElasticsearchContainer.fromSystemProperties();
+    container.start();
+  }
+
+  @Test
+  public void testChangeConfigsAndRestart() throws Exception {
+    // run connector and write
+    runSimpleTest(props);
+
+    // restart
+    props.put(BATCH_SIZE_CONFIG, "10");
+    props.put(LINGER_MS_CONFIG, "1000");
+    connect.configureConnector(CONNECTOR_NAME, props);
+
+    // write some more
+    writeRecords(10);
+    verifySearchResults(20);
+  }
+
+  @Test
+  public void testDelete() throws Exception {
+    props.put(BEHAVIOR_ON_NULL_VALUES_CONFIG, BehaviorOnNullValues.DELETE.name());
+    props.put(IGNORE_KEY_CONFIG, "false");
+    runSimpleTest(props);
+
+    // should have 10 records at this point
+    // try deleting last one
+    int lastRecord = 9;
+    connect.kafka().produce(TOPIC, String.valueOf(lastRecord), null);
+
+    // should have one less records
+    verifySearchResults(9);
+  }
+
+  @Test
+  public void testHappyPath() throws Exception {
+    runSimpleTest(props);
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void testMapping() throws Exception {
+    client.createIndices(Collections.singleton(TOPIC));
+    Schema schema = TestUtils.createSchema();
+    Mapping.createMapping(client, TOPIC, TYPE, schema);
+
+    JsonObject mapping = Mapping.getMapping(client, TOPIC, TYPE);
+    assertNotNull(mapping);
+    TestUtils.verifyMapping(client, schema, mapping);
+  }
+
+  @Test
+  public void testNullValue() throws Exception {
+    runSimpleTest(props);
+
+    // should have 10 records at this point
+    // try writing null value
+    connect.kafka().produce(TOPIC, String.valueOf(10), null);
+
+    // should still have 10 records
+    verifySearchResults(10);
+  }
+
+  /*
+   * Currently writing primitives to ES fails because ES expects a JSON document and the connector
+   * does not wrap primitives in any way into a JSON document.
+   */
+  @Test
+  public void testPrimitive() throws Exception {
+    props.put(VALUE_CONVERTER_CLASS_CONFIG, StringConverter.class.getName());
+
+    connect.configureConnector(CONNECTOR_NAME, props);
+
+    // wait for tasks to spin up
+    waitForConnectorToStart(CONNECTOR_NAME, TASKS_MAX);
+
+    for (int i  = 0; i < 10; i++) {
+      connect.kafka().produce(TOPIC, String.valueOf(i),  String.valueOf(i));
+    }
+
+    waitForRecords(0);
+  }
```

Comment:
Can we add a test for `write.method: UPSERT` as well? I believe we could've caught retry conflict errors (we saw some time ago) if we had a test for it

Reply:
that config isnt added in this branch, but I will add an IT for upsert in a subsequent PR